hack/ci-e2e.sh

```diff
@@ -189,6 +198,10 @@ for overlay in "${IRONIC_OVERLAYS[@]}"; do
   "${overlay}/ironic-auth-config"
 done
 
+sed -i "s|SSH_PUB_KEY_CONTENT|${pub_ssh_key}|" "${REPO_ROOT}"/test/e2e/data/ironic-standalone-operator/ironic/base/ironic.yaml
+sed -i "s/.*username.*/  - username=${IRONIC_USERNAME}/" "${REPO_ROOT}"/test/e2e/data/ironic-standalone-operator/components/basic-auth/kustomization.yaml
+sed -i "s/.*password.*/  - password=${IRONIC_PASSWORD}/" "${REPO_ROOT}"/test/e2e/data/ironic-standalone-operator/components/basic-auth/kustomization.yaml
```

Comment:
This is not so nice, because it leaves the repo in dirty state.
There isn't much we can do about the public ssh key here, but the username and password would be better to create from files in my opinion. Those file should then be added to .gitignore.